src/ui_basic/panel.cc

```diff
@@ -143,6 +143,35 @@ void Panel::free_children() {
 	first_child_ = nullptr;
 }
 
+// static
+void* Panel::runthread(void* p) {
```

Comment:
Why not a Panel* ? please provide docs, otherwise  mutlithreading feels like dark magic ;-)